File: couchbase-entity/src/main/java/com/kaufland/util/ElementUtil.java
package com.kaufland.util;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;
import java.util.List;

/**
 * Created by sbra0902 on 26.05.17.
 */

public class ElementUtil {

    public static List<String> splitGenericIfNeeded(String name){


        List<String> result = new ArrayList<>();
        if(name.contains("<")){

            result.add(name.substring(0, name.indexOf("<")).trim());

            for (String item : Arrays.asList(name.substring(name.indexOf("<") + 1, name.indexOf(">")).split(","))){
                result.add(item.trim());
            }
        }else{
            result.add(name);
        }

        return result;

    }

}

File: crystal-map-processor/src/main/java/com/schwarz/crystalprocessor/util/ElementUtil.kt
package com.schwarz.crystalprocessor.util

import java.util.ArrayList
import java.util.Arrays

object ElementUtil {
    fun splitGenericIfNeeded(name: String): List<String> {
        val result = ArrayList<String>()
        if (name.contains("<")) {
            result.add(name.substring(0, name.indexOf("<")).trim { it <= ' ' })

            for (item in Arrays.asList(
                *name
                    .substring(
                        name.indexOf("<") + 1,
                        name.indexOf(">"),
                    ).split(",".toRegex())
                    .dropLastWhile {
                        it.isEmpty()
                    }.toTypedArray(),
            )) {
                result.add(item.trim { it <= ' ' })
            }
        } else {
            result.add(name)
        }

        return result
    }
}
